Return time in POSIXct for all functions

PaulineNimo opened the pull request.

PaulineNimo committed:
Additional modify, status and cancel functions using client ID

PaulineNimo committed:
Updating ftx historical prices

PaulineNimo committed:
Updating ftx trades

PaulineNimo committed:
Updating ftx future markets

PaulineNimo committed:
Updating ftx_future_stat

PaulineNimo committed:
Updating ftx future funding rates

PaulineNimo committed:
Updating ftx open orders

PaulineNimo committed:
Updating ftx orders history

PaulineNimo committed:
Updating ftx place order

PaulineNimo committed:
Updating ftx modify order

PaulineNimo committed:
Updating ftx order status

PaulineNimo committed:
Updating ftx modify order clientid

PaulineNimo committed:
Updating ftx order status clientid

PaulineNimo committed:
Updating order fills

PaulineNimo committed:
Updating ftx funding payments

PaulineNimo committed:
Updating ftx spot lending history

PaulineNimo committed:
Updating ftx spot borrow history

andreskull commented:
1. Check that every function with start time and end time will be start_time and end_time optional (start_time=NA, end_time=NA).
2. Add tz="GMT" parameter for all functions that return time and pass tz parameter to as.POSIXct function.
3. In case the resulting tibble is empty you cannot mutate the time. This will throw nasty error. Mutate only if there is something.

andreskull commented:
Try to encapsulate duplicate code in the functions to separate helper functions to make changes better managable.

andreskull commented:
> 1. Check that every function with start time and end time will be start_time and end_time optional (start_time=NA, end_time=NA).

This should solve the issue: https://github.com/andreskull/rFTX/issues/5

andreskull commented:
> > 1. Check that every function with start time and end time will be start_time and end_time optional (start_time=NA, end_time=NA).
> 
> This should solve the issue: #5

Sorry, the ftx_my_spot_borrow_history() function does not have the start_time and end_time parameter actually. Should be added.

andreskull commented:
> > 1. Check that every function with start time and end time will be start_time and end_time optional (start_time=NA, end_time=NA).
> 
> This should solve the issue: #5

Actually, it does not solve the issue with this function.
I made the change locally and the result is the following:
```
> ftx_my_spot_borrow_history(key, secret, start_time = now() - days(200), end_time = now()) 
2021-11-02 21:03:37 ERROR::Not logged in
Error in x[[1]] : subscript out of bounds
In addition: Warning message:
In sprintf("Not logged in", query = "1618596217.50369,1635879817.5057") :
 
 Error in x[[1]] : subscript out of bounds 
```
 
 Needs more investigations.

andreskull commented:
4. start_time and end_time should be converted using as.integer() function instead of as.numeric(). This avoids the floating numbers in the query as seen here https://github.com/andreskull/rFTX/pull/4#issuecomment-958096297

PaulineNimo committed:
every function with start time and end time to be start_time and end_time optional

PaulineNimo committed:
Add tz="GMT" parameter for all functions that return time and pass tz parameter to as.POSIXct function.

PaulineNimo committed:
start_time and end_time should be converted using as.integer() function instead of as.numeric()

PaulineNimo committed:
Adding helper function and mutate only if tibble contains column

PaulineNimo committed:
Replacing repeated code with result formatter helper function

PaulineNimo committed:
Resolving issue with time zone argument in result formatter helper function

PaulineNimo committed:
Updating Roxygen comments on timezone update

PaulineNimo committed:
Updating ftx coin markets to use helper function

PaulineNimo committed:
Updating ftx positions to use helper function

PaulineNimo committed:
Updating ftx future markets with helper function

PaulineNimo committed:
Updating functions to use format helper

PaulineNimo committed:
Updating result formatter to use format helper

andreskull committed:
ftx_orderbook: loginfo --> logerror
ftx_historical_prices: select relevant columns and leave out confusing time column
